M2/Macaulay2/e/eigen.cpp

```diff
@@ -0,0 +1,216 @@
+#include <cstdlib>
+
+#if 0
+//void* my_malloc( std::size_t size );
+//void my_free( void* ptr );
+namespace std {
+void* my_malloc( std::size_t size );
+void my_free( void* ptr );
+}
+using std::my_malloc;
+using std::my_free;
+
+#define malloc my_malloc
+#define free my_free
+#endif
+
+#define mpfr foo
+#include "mpreal.h"
+#include <unsupported/Eigen/MPRealSupport>
+#undef mpfr
```

Comment:
Can you add a comment about what is happening here?

Reply:
@mikestillman -- this is for you

Reply:
I'll look into this with @antonleykin